packages/desktopjs-openfin/src/openfin.ts

```diff
@@ -673,13 +674,15 @@ export class OpenFinContainer extends WebContainerBase {
     public buildLayout(): Promise<PersistedWindowLayout> {
         const layout = new PersistedWindowLayout();
 
+        // eslint-disable-next-line no-async-promise-executor
         return new Promise<PersistedWindowLayout>(async (resolve, reject) => {
             const windows = await this.getAllWindows();
             const mainWindow = this.getMainWindow();
             const promises: Promise<void>[] = [];
 
             windows.filter(window => window.name !== "queueCounter" && !window.name.startsWith(OpenFinContainer.notificationGuid))
                 .forEach(djsWindow => {
+                    // eslint-disable-next-line no-async-promise-executor
```

Comment:
No expect when it comes to the async promise executor but looking at this https://eslint.org/docs/rules/no-async-promise-executor it seems like the code below wants refactoring

Reply:
It looks like many of the functions (window.getBounds, getOptions, getGroup) are using a callback pattern that we could easily change/wrap to return a Promise.
I'll take a look at these later. Maybe I'll add an issue to refactor all async code, since I think the async/await style is much more readable.

Reply:
when we upgrade to openfin v2 api we can switch to async/await as they made the change themselves.  v1 is callback based.